src/components/Main/MainView/MessageInput/DropdownSuggester.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div
+    :class="$style.container"
+    :style="{ top: position.top - 152 + 'px', left: position.left + 60 + 'px' }"
+  >
+    <div
+      :class="$style.item"
+      v-for="(candidate, index) in candidates"
+      :key="index"
+      @click="onClick(e, candidate)"
+    >
+      {{ candidate }}
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, SetupContext, PropType } from 'vue'
+import Icon from '@/components/UI/Icon.vue'
+
+const useClickHandlers = (context: SetupContext<'select'[]>) => {
+  const onClick = (_e: MouseEvent, word: string) => {
+    context.emit('select', word)
+  }
+  return { onClick }
+}
+
+// スタンプのプレビュー表示とか用(wordCompleterからexportしたほうが良さそう)
+// interface Candidate {
+//   text: string
+//   type:
+// }
+
+export default defineComponent({
+  name: 'DropdownSuggester',
+  // components: {
+  //   Icon
+  // },
+  props: {
+    // candidates: {
+    //   type: Array as PropType<Candidate[]>,
+    //   default: null
+    // }
+    candidates: {
+      type: Array as PropType<string[]>,
+      default: []
+    },
+    position: {
+      type: Object as PropType<{ top: number; left: number }>
+    }
+  },
+  emits: ['select'],
```

Comment:
書くなら
```suggestion
  emits: {
    select: () => true
  }
```
みたいに引数明示したい 🙏 
https://v3.vuejs.org/guide/typescript-support.html#annotating-emits